src/instructlab/eval/mt_bench_branch_generator.py

```diff
@@ -22,7 +22,13 @@
 def get_file_paths(directory):
     logger.debug(locals())
     file_paths = []
-    for root, _, files in os.walk(directory):
+    for root, dirs, files in os.walk(directory):
+        if root == directory and "knowledge" in dirs:
+            logger.debug(
+                "skipping knowledge directory: %s", os.path.join(root, "knowledge")
+            )
+            dirs.remove("knowledge")
```

Comment:
What's the purpose of removing "knowledge" here from `dirs`? I don't see `dirs` being used again in the function.

Reply:
It's how os.walk is designed. As it iterates, if you remove a directory from `dirs` it does not descend into that dir. Better than going through every file in that tree and having to check its path.